src/components/materialization/Edit.tsx

```diff
@@ -56,57 +55,28 @@ function MaterializationEdit() {
     const resetDetailsFormState = useDetailsForm_resetState();
 
     // Draft Editor Store
-    const draftId = useZustandStore<
-        EditorStoreState<DraftSpecQuery>,
-        EditorStoreState<DraftSpecQuery>['id']
-    >(draftEditorStoreName, (state) => state.id);
-
-    const setDraftId = useZustandStore<
-        EditorStoreState<DraftSpecQuery>,
-        EditorStoreState<DraftSpecQuery>['setId']
-    >(draftEditorStoreName, (state) => state.setId);
+    const draftId = useEditorStore_id();
+    const setDraftId = useEditorStore_setId();
 
     // Endpoint Config Store
     const endpointConfigErrorsExist = useEndpointConfigStore_errorsExist();
     const resetEndpointConfigState = useEndpointConfigStore_reset();
     const endpointConfigChanged = useEndpointConfigStore_changed();
 
     // Form State Store
-    const messagePrefix = useZustandStore<
-        EntityFormState,
-        EntityFormState['messagePrefix']
-    >(formStateStoreName, (state) => state.messagePrefix);
-
-    const setFormState = useZustandStore<
-        EntityFormState,
-        EntityFormState['setFormState']
-    >(formStateStoreName, (state) => state.setFormState);
-
-    const resetFormState = useZustandStore<
-        EntityFormState,
-        EntityFormState['resetState']
-    >(formStateStoreName, (state) => state.resetState);
-
-    const exitWhenLogsClose = useZustandStore<
-        EntityFormState,
-        EntityFormState['formState']['exitWhenLogsClose']
-    >(formStateStoreName, (state) => state.formState.exitWhenLogsClose);
+    const messagePrefix = useFormStateStore_messagePrefix();
 
-    // Resource Config Store
-    const resourceConfigErrorsExist = useZustandStore<
-        ResourceConfigState,
-        ResourceConfigState['resourceConfigErrorsExist']
-    >(resourceConfigStoreName, (state) => state.resourceConfigErrorsExist);
+    const setFormState = useFormStateStore_setFormState();
+
+    const resetFormState = useFormStateStore_resetState();
 
-    const resetResourceConfigState = useZustandStore<
-        ResourceConfigState,
-        ResourceConfigState['resetState']
-    >(resourceConfigStoreName, (state) => state.resetState);
+    const exitWhenLogsClose = useFormStateStore_exitWhenLogsClose();
+
+    // Resource Config Store
+    // const resourceConfigChanged = useResourceConfig_stateChanged();
```

Comment:
I would like to address this in a separate PR because it is related to a larger code _quirk_ of the create and edit workflows. Given both forms involve a degree of prefilled data, the form data is always in a _changed_ state and thus the data loss modal is prompted on any exit.